multimodal/src/autogluon/multimodal/models/mmocr_text_detection.py

```diff
@@ -0,0 +1,173 @@
+    from mmocr.models import build_detector
+    from mmocr.utils.model import revert_sync_batchnorm
+except ImportError:
+    mmocr = None
+
+try:
+    import mmdet
+    from mmdet.core import get_classes
+except ImportError:
+    mmdet = None
+
+from ..constants import AUTOMM, BBOX, COLUMN, COLUMN_FEATURES, FEATURES, IMAGE, IMAGE_VALID_NUM, LABEL, LOGITS, MASKS
+from .utils import assign_layer_ids, get_column_features, get_model_head
+
+logger = logging.getLogger(AUTOMM)
+
+
+class MMOCRAutoModelForTextDetection(nn.Module):
+    """
+    Support MMOCR object detection models.
+    Refer to https://github.com/open-mmlab/mmdetection
+    """
+
+    def __init__(
+        self,
+        prefix: str,
+        checkpoint_name: str,
+        num_classes: Optional[int] = None,
+        pretrained: Optional[bool] = True,
+    ):
+        """
+        Load a pretrained object detector from MMdetection.
+
+        Parameters
+        ----------
+        prefix
+            The prefix of the MMdetAutoModelForObjectDetection model.
+        checkpoint_name
+            Name of the mmdet checkpoint.
+        num_classes
+            The number of classes.
+        pretrained
+            Whether using the pretrained mmdet models. If pretrained=True, download the pretrained model.
+        """
+        super().__init__()
+        logger.debug(f"initializing {checkpoint_name}")
+        self.checkpoint_name = checkpoint_name
+        self.pretrained = pretrained
+
+        # download config and checkpoint files using openmim
+        checkpoints = download(package="mmocr", configs=[checkpoint_name], dest_root=".")
+
+        # read config files
+        assert mmcv is not None, "Please install mmcv-full by: mim install mmcv-full."
+        config_file = checkpoint_name + ".py"
+        if isinstance(config_file, str):
+            self.config = mmcv.Config.fromfile(config_file)
+
+        # build model and load pretrained weights
+        assert mmocr is not None, "Please install MMOCR by: pip install mmocr."
+        # previous version1
+        # print(self.config)
+        # self.model = MMOCR(det_config=config_file, det_ckpt=checkpoints[0],recog=None)
+        # self.model.cfg = self.config  # save the config in the model for convenience
+        # self.prefix = prefix
+
+        # previous version2
+        # self.model = init_detector(config_file, checkpoints[0])
+
+        checkpoint = checkpoints[0]
+        self.model = build_detector(self.config.model, test_cfg=self.config.get("test_cfg"))
+        if checkpoint is not None:
+            checkpoint = load_checkpoint(self.model, checkpoint, map_location="cpu")
+            if "CLASSES" in checkpoint.get("meta", {}):
+                self.model.CLASSES = checkpoint["meta"]["CLASSES"]
+            else:
+                warnings.simplefilter("once")
+                warnings.warn(
+                    "Class names are not saved in the checkpoint's " "meta data, use COCO classes by default."
```

Comment:
Is it still COCO? I think we need to change the assert message as well.

Reply:
Actually it depends on the dataset used. I will delete these lines.